pandas/tests/arrays/test_datetimelike.py

```diff
@@ -149,8 +149,25 @@ def test_array(self, tz_naive_fixture):
         assert result.base is expected.base
         assert result.base is not None
 
+    def test_array2(self, datetime_index):
+        arr = DatetimeArray(datetime_index)
+
+        result = np.asarray(arr)
+        expected = arr._data
+        assert result is expected
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = np.asarray(arr, dtype=object)
+        expected = np.array(list(arr), dtype=object)
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = np.asarray(arr, dtype='int64')
```

Comment:
what about other dtypes?

Reply:
DatetimeArray only overrides for object dtype. NumPy takes care of the other dtypes, so I don't think there's anything additional to test, is there?